### PR TITLE
ARROW-17888: [Docs] Add reference of the cookbook contrib page to New Contributor's Guide

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR adds a reference for Apache Arrow Cookbook contributing info page to the New Contributor's Guide.

Screenshot 2022-09-30 at 15 27 21

### CODE REVIEW

**Comment by AlenkaF** [comment]
@ianmcook @thisisnic would you mind having a look at this tiny PR?

File: docs/source/developers/guide/index.rst [hunk 1]
```diff
@@ -147,11 +147,23 @@ of adding a basic feature.
...
+============================
 
-**We want to encourage everyone to contribute to Arrow!**
+There are lots of ways to contribute to the project besides writing code!
+
+* Improving the **documentation** is a great way to start contributing!
+  For more information visit :ref:`documentation` section of the guide.
+
+* **Apache Arrow Cookbooks** are a collection of recipes presenting different
+  functionalities of Apache Arrow project. They are also a great way to start
```
Comment (thisisnic) [inline docs/source/developers/guide/index.rst:159]:
```suggestion
* **Apache Arrow Cookbooks** are a collection of recipes for solving various problems 
and completing different tasks using Apache Arrow. They are also a great way to start
```
Small change here, just to emphasise that the recipes are focussed on practical tasks.

**Comment by thisisnic** [comment]
Thanks for adding this! Just one tiny phrasing suggestion, otherwise LGTM!

File: docs/source/developers/guide/index.rst [hunk 2]
```diff
@@ -147,11 +147,23 @@ of adding a basic feature.
...
+  contributing. For more information visit
+  `How to contribute to Apache Arrow Cookbook <https://github.com/apache/arrow-cookbook/blob/main/CONTRIBUTING.md>`_
+  located in the Apache Arrow Cookbook repository.
 
+You can also welcome take a look at :ref:`other-resources` section.
```
Comment (raulcd) [inline docs/source/developers/guide/index.rst:164]: I am not a native English speaker, so feel free to disregard:
```suggestion
You are also welcome to take a look at :ref:`other-resources` section.
```

**Review by raulcd — COMMENTED** [review]
LGTM, minor grammar change

**Comment by AlenkaF** [comment]
Thank you for the comments!
Will wait for the CI and then merge if green.

**Pull request merged**